sources/api/prairiedog/build.rs

```diff
@@ -0,0 +1,54 @@
+// Automatically generate README.md from rustdoc.
+
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use std::{env, process};
+
+fn main() {
+    // The code below emits `cfg` operators to conditionally compile this program based on the
+    // current variant.
+    // TODO: Replace this approach when the build system supports ideas like "variant
+    // tags": https://github.com/bottlerocket-os/bottlerocket/issues/1260
```

Comment:
It's mostly an issue with the compiler. Only metal variants right now have `boot: BootSettings` in its setting struct. In `prairiedog::generate_boot_config`, I need that field for fetching the settings to serializing the bootconfig.

So if I try building a variant that doesn't have `BootSettings` in its settings struct, then I run into this:
```
error[E0609]: no field `boot` on type `Settings`
  --> api/prairiedog/src/bootconfig.rs:57:25
   |
57 |         .and_then(|s| s.boot)
   |                         ^^^^ unknown field
   |
   = note: available fields are: `motd`, `kubernetes`, `updates`, `host_containers`, `bootstrap_containers` ... and 9 others
```

I could also add that field to all variant's settings struct. But that would mean it would be an inert setting that won't do anything.

Reply:
Ok, I think I can avoid this error by doing `apiclient::raw_request` and going through the JSON response body and check via `serde_json::Value` and see if `boot` exists under `settings` or not. That would avoid all the conditional compilation and we'd just output an error message for boot configuration subcommands in variants that don't support it.